invenio_rdm_records/services/results.py

```diff
@@ -69,3 +69,32 @@ def get_value_service(self, value):
     def pick(self, identity, resolved_rec):
         """Pick fields defined in the entity resolver."""
         return resolved_rec
+
+class RDMRecordList(RecordList):
```

Comment:
The issue was that drafts are searched through the records service, which in turn uses the default `RecordList` class to project the search results. 

The default implementation uses `self._service.record_cls` unconditionally, which in turn projects the result always to a record. This is not correct if the service shares the search for records and drafts. For instance, an unpublished draft always had the attribute `is_draft=False` because the attribute `is_draft` of a `Record` is always set to `False`.